### DISCUSSION TITLE
GitHub checks in combination with allowed-to-fail GitHub Action

### DISCUSSION THREAD

**Discussion by DeepSpace2** [community/community#27069] [Actions]
<p>When using an allowed-to-fail GitHub Action and it does fail (who figured :slight_smile:) the GitHub check icon is a red cross. I’d have hoped that in case an allowed-to-fail Action (or Actions) is the only failing job we would get a different feedback (maybe a yellow-greenish half-check mark or something along these lines).</p>

**Comment by joshmgross** [comment]
<p>:wave: Hey <a class="mention" href="/deepspace2">@DeepSpace2</a>,</p>
<p>Could you describe your workflow and how you’re allowing an action to fail?</p>
<p>With a step like:</p>
<pre><code class="lang-yaml">    steps:
      - run: exit 1
        continue-on-error: true
</code></pre>
<p>The job itself still succeeds.</p>
<p>Example: <a href="https://github.com/joshmgross/actions-testing/actions/runs/940903240" class="inline-onebox">Create continue-on-error.yml · joshmgross/actions-testing@1b25d80 · GitHub</a></p>
<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error</a></p>

**Comment by DeepSpace2** [comment]
<p>Hi. This is my entire workflow <code>.yml</code> file:</p>
<pre><code class="lang-auto">name: test

on:
  push:
  pull_request:

jobs:
  build:
    runs-on: ubuntu-latest
    continue-on-error: ${{ matrix.experimental }}
    name: Python ${{ matrix.python-version }}
    strategy:
      fail-fast: false
      matrix:
        python-version: [3.6, 3.7, 3.8, 3.9]
        experimental: [false]
        include:
          - python-version: 3.10-dev
            experimental: true

    steps:
      - uses: actions/checkout@v2
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v2
        if: "!endsWith(matrix.python-version, '-dev')"
        with:
          python-version: ${{ matrix.python-version }}
      - name: Set up Python ${{ matrix.python-version }}
        uses: deadsnakes/action@v1.0.0
        if: endsWith(matrix.python-version, '-dev')
        with:
          python-version: ${{ matrix.python-version }}
      - name: Installing dependencies Python ${{ matrix.python-version }}
        run: |
          pip install -r requirements.txt
          pip install codecov coverage
      - name: Running tests Python ${{ matrix.python-version }}
        run: |
          coverage run -m unittest discover
          coverage xml
      - name: Uploading coverage Python ${{ matrix.python-version }}
        uses: codecov/codecov-action@v1
</code></pre>
<p>The problem is not that the workflow fails. The problem is that the icon is a red cross, which is a misleading indication.</p>

**Comment by joshmgross** [comment, marked as answer]
<p>If you set <code>continue-on-error</code> for the specific <code>step</code> that you expect to fail, the job itself should show as a success.</p>

**Reply by berezovskyi** [reply]
But this will make the failing matrix job appear to pass. Is there a way instead to make the matrix pass (PR and merge check green) yet still show individual matrix job failures in the details?
